components/device-mgt/org.wso2.carbon.device.mgt.extensions/src/main/java/org/wso2/carbon/device/mgt/extensions/device/type/template/util/DeviceTypeUtils.java

```diff
@@ -101,25 +102,29 @@ public static void setupDeviceManagementSchema(String datasourceName, String dev
     public static Registry getConfigurationRegistry() throws DeviceTypeMgtPluginException {
         try {
             int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
-            return DeviceTypeExtensionDataHolder.getInstance().getRegistryService()
-                    .getConfigSystemRegistry(tenantId);
+            org.wso2.carbon.registry.core.service.RegistryService registryService = DeviceTypeExtensionDataHolder
+                    .getInstance().getRegistryService();
+            return registryService == null ? null : registryService.getConfigSystemRegistry(tenantId);
```

Comment:
Don't we need to throw the exception when the resgistry service is null? Can we do any succesful operation if the registry service is null here?